tests/ipsecondary_test.py

```diff
@@ -0,0 +1,537 @@
+    loglevel = 1
+
+    '''
+
+    SECONDARY_CONFIG_TEMPLATE = '''
+    {{
+      "IP": {{
+        "secondaries_wait_port": {port},
+        "secondaries_wait_timeout": {timeout},
+        "secondaries": [] 
+      }}
+    }}
+    '''
+
+    @property
+    def id(self):
+        return self._id
+
+    def __init__(self, aktualizr_primary_exe, aktualizr_info_exe, id,
+                 server_url, ca, pkey, cert, wait_port=9040, wait_timeout=60, secondary=None, output_logs=True):
+        self._aktualizr_primary_exe = aktualizr_primary_exe
+        self._aktualizr_info_exe = aktualizr_info_exe
+        self._id = id
+        self._storage_dir = tempfile.TemporaryDirectory()
+
+        with open(path.join(self._storage_dir.name, 'secondary_config.json'), 'w+') as secondary_config_file:
+            secondary_cfg = json.loads(Aktualizr.SECONDARY_CONFIG_TEMPLATE.format(port=wait_port, timeout=wait_timeout))
+            json.dump(secondary_cfg, secondary_config_file)
+            self._secondary_config_file = secondary_config_file.name
+
+        with open(path.join(self._storage_dir.name, 'config.toml'), 'w+') as config_file:
+            config_file.write(Aktualizr.CONFIG_TEMPLATE.format(server_url=server_url,
+                                                               ca_path=ca, pkey_path=pkey, cert_path=cert,
+                                                               serial=id[1], hw_ID=id[0],
+                                                               storage_dir=self._storage_dir,
+                                                               db_path=path.join(self._storage_dir.name, 'db.sql'),
+                                                               secondary_cfg_file=self._secondary_config_file))
+            self._config_file = config_file.name
+
+        self.add_secondary(secondary) if secondary else None
+        self._output_logs = output_logs
+
+    def add_secondary(self, secondary):
+        with open(self._secondary_config_file, "r+") as config_file:
+            sec_cfg = json.load(config_file)
+            sec_cfg["IP"]["secondaries"].append({"addr": "127.0.0.1:{}".format(secondary.port)})
+            config_file.seek(0)
+            json.dump(sec_cfg, config_file)
+
+    def update_wait_timeout(self, timeout):
+        with open(self._secondary_config_file, "r+") as config_file:
+            sec_cfg = json.load(config_file)
+            sec_cfg["IP"]["secondaries_wait_timeout"] = timeout
+            config_file.seek(0)
+            json.dump(sec_cfg, config_file)
+
+    def run(self, run_mode):
+        subprocess.run([self._aktualizr_primary_exe, '-c', self._config_file, '--run-mode', run_mode], check=True)
+
+    def get_info(self):
+        return str(subprocess.check_output([self._aktualizr_info_exe, '-c', self._config_file]))
+
+    # ugly stuff that could be removed if Aktualizr had exposed API to check status
+    # or aktializr-info had output status/info in a structured way (e.g. json)
```

Comment:
I would not at all be opposed to cleaning up the output of aktualizr-info. I would keep that type of thing out of aktualizr itself, though.

Reply:
ok. I'll elaborate my opinion during the stand-up meeting or via email.